### PR TITLE
[k8s-helm-windows] fix bugs and add Service

### PR DESCRIPTION
**Pull request by povilasv**
# Description

[SC-18267]

* Add Service for DaemonSet, since windows does not support hostnetworking
* Fix includeCollectorLogs=true breaks configuration.
* Add support for Deployment and StatefulSet options.

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the relevant Helm chart(s) version(s)
- [ ] I have updated the relevant component changelog(s)
- [ ] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

File: otel-agent/k8s-helm-windows/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: opentelemetry-coralogix-windows
-version: 0.0.1
+version: 0.0.2
```
Comment (matej-g) [inline otel-agent/k8s-helm-windows/Chart.yaml:3]: I guess we'll need to add another CI check for the new chart, but I'm happy to do it as a follow up once we also have images published.

File: otel-agent/k8s-helm-windows/templates/service.yaml [hunk 2]
```diff
@@ -27,7 +26,7 @@ spec:
     {{- end }}
   {{- end }}
   ports: {{ include "opentelemetry-collector.deploymentPortsConfig" . | nindent 4 }}
+  internalTrafficPolicy: {{ include "opentelemetry-collector.serviceInternalTrafficPolicy" . }}
```
Comment (matej-g) [inline otel-agent/k8s-helm-windows/templates/service.yaml:29]: Is this change Windows related / required for Windows?

**Review by matej-g — APPROVED** [review]
Looks good to me 👍, just one clarification.

**Review by oded-dd — APPROVED** [review]
